File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mr/knowledgetype/d/KnowledgeTypeDO.java
package iih.ci.mr.knowledgetype.d;

import xap.sys.appfw.orm.desc.DescManager;
import xap.mw.coreitf.c.GlobalConst;
import xap.mw.coreitf.i.IDODesc;
import xap.mw.coreitf.d.*;
import xap.mw.core.data.*;
import iih.ci.mr.knowledgetype.d.desc.KnowledgeTypeDODesc;
import java.math.BigDecimal;

/**
 * 知识库类型 DO数据 
 * 
 */
public class KnowledgeTypeDO extends BaseDO {
	private static final long serialVersionUID = 1L;

	public static final String ID_KNOWLEDGE_TYPE= "Id_knowledge_type";
	public static final String ID_GRP= "Id_grp";
	public static final String ID_ORG= "Id_org";
	public static final String SORTNO= "Sortno";
	public static final String CODE= "Code";
	public static final String NAME= "Name";
	public static final String PYCODE= "Pycode";
	public static final String WBCODE= "Wbcode";
	public static final String MNECODE= "Mnecode";
	public static final String ID_DEPT= "Id_dept";
	public static final String ID_USER= "Id_user";
	public static final String PERSONAL= "Personal";
	public static final String ADMINISTRATIVE= "Administrative";
	public static final String ATTRIBUTE= "Attribute";
	public static final String DES= "Des";
	public static final String CREATEDBY= "Createdby";
	public static final String CREATEDTIME= "Createdtime";
	public static final String MODIFIEDBY= "Modifiedby";
	public static final String MODIFIEDTIME= "Modifiedtime";
	public static final String ORG_CODE= "Org_code";
	public static final String ORG_NAME= "Org_name";
	public static final String DEPT_CODE= "Dept_code";
	public static final String DEPT_NAME= "Dept_name";
	public static final String USER_NAME= "User_name";
	public static final String USER_CODE= "User_code";
	public static final String DS = "Ds";
	public static final String SV = "Sv";
	
	public String getId_knowledge_type() {
		return ((String) getAttrVal("Id_knowledge_type"));
	}	
	public void setId_knowledge_type(String Id_knowledge_type) {
		setAttrVal("Id_knowledge_type", Id_knowledge_type);
	}
	public String getId_grp() {
		return ((String) getAttrVal("Id_grp"));
	}	
	public void setId_grp(String Id_grp) {
		setAttrVal("Id_grp", Id_grp);
	}
	public String getId_org() {
		return ((String) getAttrVal("Id_org"));
	}	
	public void setId_org(String Id_org) {
		setAttrVal("Id_org", Id_org);
	}
	public Integer getSortno() {
		return ((Integer) getAttrVal("Sortno"));
	}	
	public void setSortno(Integer Sortno) {
		setAttrVal("Sortno", Sortno);
	}
	public String getCode() {
		return ((String) getAttrVal("Code"));
	}	
	public void setCode(String Code) {
		setAttrVal("Code", Code);
	}
	public String getName() {
		return ((String) getAttrVal("Name"));
	}	
	public void setName(String Name) {
		setAttrVal("Name", Name);
	}
	public String getPycode() {
		return ((String) getAttrVal("Pycode"));
	}	
	public void setPycode(String Pycode) {
		setAttrVal("Pycode", Pycode);
	}
	public String getWbcode() {
		return ((String) getAttrVal("Wbcode"));
	}	
	public void setWbcode(String Wbcode) {
		setAttrVal("Wbcode", Wbcode);
	}
	public String getMnecode() {
		return ((String) getAttrVal("Mnecode"));
	}	
	public void setMnecode(String Mnecode) {
		setAttrVal("Mnecode", Mnecode);
	}
	public String getId_dept() {
		return ((String) getAttrVal("Id_dept"));
	}	
	public void setId_dept(String Id_dept) {
		setAttrVal("Id_dept", Id_dept);
	}
	public String getId_user() {
		return ((String) getAttrVal("Id_user"));
	}	
	public void setId_user(String Id_user) {
		setAttrVal("Id_user", Id_user);
	}
	public FBoolean getPersonal() {
		return ((FBoolean) getAttrVal("Personal"));
	}	
	public void setPersonal(FBoolean Personal) {
		setAttrVal("Personal", Personal);
	}
	public FBoolean getAdministrative() {
		return ((FBoolean) getAttrVal("Administrative"));
	}	
	public void setAdministrative(FBoolean Administrative) {
		setAttrVal("Administrative", Administrative);
	}
	public Integer getAttribute() {
		return ((Integer) getAttrVal("Attribute"));
	}	
	public void setAttribute(Integer Attribute) {
		setAttrVal("Attribute", Attribute);
	}
	public String getDes() {
		return ((String) getAttrVal("Des"));
	}	
	public void setDes(String Des) {
		setAttrVal("Des", Des);
	}
	public String getCreatedby() {
		return ((String) getAttrVal("Createdby"));
	}	
	public void setCreatedby(String Createdby) {
		setAttrVal("Createdby", Createdby);
	}
	public FDateTime getCreatedtime() {
		return ((FDateTime) getAttrVal("Createdtime"));
	}	
	public void setCreatedtime(FDateTime Createdtime) {
		setAttrVal("Createdtime", Createdtime);
	}
	public String getModifiedby() {
		return ((String) getAttrVal("Modifiedby"));
	}	
	public void setModifiedby(String Modifiedby) {
		setAttrVal("Modifiedby", Modifiedby);
	}
	public FDateTime getModifiedtime() {
		return ((FDateTime) getAttrVal("Modifiedtime"));
	}	
	public void setModifiedtime(FDateTime Modifiedtime) {
		setAttrVal("Modifiedtime", Modifiedtime);
	}
	public String getOrg_code() {
		return ((String) getAttrVal("Org_code"));
	}	
	public void setOrg_code(String Org_code) {
		setAttrVal("Org_code", Org_code);
	}
	public String getOrg_name() {
		return ((String) getAttrVal("Org_name"));
	}	
	public void setOrg_name(String Org_name) {
		setAttrVal("Org_name", Org_name);
	}
	public String getDept_code() {
		return ((String) getAttrVal("Dept_code"));
	}	
	public void setDept_code(String Dept_code) {
		setAttrVal("Dept_code", Dept_code);
	}
	public String getDept_name() {
		return ((String) getAttrVal("Dept_name"));
	}	
	public void setDept_name(String Dept_name) {
		setAttrVal("Dept_name", Dept_name);
	}
	public String getUser_name() {
		return ((String) getAttrVal("User_name"));
	}	
	public void setUser_name(String User_name) {
		setAttrVal("User_name", User_name);
	}
	public String getUser_code() {
		return ((String) getAttrVal("User_code"));
	}	
	public void setUser_code(String User_code) {
		setAttrVal("User_code", User_code);
	}

	public Integer getDs() {
		return ((Integer) getAttrVal(GlobalConst.DELETE_SIGN));
	}
	
	public void setDs(Integer ds) {
		setAttrVal(GlobalConst.DELETE_SIGN, ds);
	}
	
	public FDateTime getSv() {
		return ((FDateTime) getAttrVal(GlobalConst.SYSTEM_VERSION));
	}
	
	public void setSv(FDateTime sv) {
		setAttrVal(GlobalConst.SYSTEM_VERSION, sv);
	}

//	@Override
//	public java.lang.String getParentPKFieldName() {
//		return null;
//	}
  
	@Override
	public String getPKFieldName() {
		return "Id_knowledge_type";
	}
	
	@Override
	public String getTableName() {	  
		return "ci_mr_knowledge_type";
	}
	
	@Override
	public IDODesc getDODesc() {
		return DescManager.getInstance().getDODesc(KnowledgeTypeDODesc.class);
	}
	
}

File: client/iih_stub/iih.ci_stub/iih.ci.mr/knowledgetype/d/KnowledgeTypeDO.cs

using System;
using xap.mw.core.data;
using xap.mw.coreitf.d;
using System.Collections.Generic;

namespace iih.ci.mr.knowledgetype.d
{
    /// <summary>
    /// KnowledgeTypeDO 的摘要说明。
    /// </summary>
    public class KnowledgeTypeDO : BaseDO {

		public const string TABLE_NAME = "ci_mr_knowledge_type";
		public const string TABLE_ALIAS_NAME = "a0";

        public KnowledgeTypeDO() {
        }
		public string Id_knowledge_type {
            get { return getAttrVal<string>("Id_knowledge_type",null); }
            set { setAttrVal<string>("Id_knowledge_type", value); }
        }
		public string Id_grp {
            get { return getAttrVal<string>("Id_grp",null); }
            set { setAttrVal<string>("Id_grp", value); }
        }
		public string Id_org {
            get { return getAttrVal<string>("Id_org",null); }
            set { setAttrVal<string>("Id_org", value); }
        }
		public int? Sortno {
            get { return getAttrVal<int?>("Sortno",null); }
            set { setAttrVal<int?>("Sortno", value); }
        }
		public string Code {
            get { return getAttrVal<string>("Code",null); }
            set { setAttrVal<string>("Code", value); }
        }
		public string Name {
            get { return getAttrVal<string>("Name",null); }
            set { setAttrVal<string>("Name", value); }
        }
		public string Pycode {
            get { return getAttrVal<string>("Pycode",null); }
            set { setAttrVal<string>("Pycode", value); }
        }
		public string Wbcode {
            get { return getAttrVal<string>("Wbcode",null); }
            set { setAttrVal<string>("Wbcode", value); }
        }
		public string Mnecode {
            get { return getAttrVal<string>("Mnecode",null); }
            set { setAttrVal<string>("Mnecode", value); }
        }
		public string Id_dept {
            get { return getAttrVal<string>("Id_dept",null); }
            set { setAttrVal<string>("Id_dept", value); }
        }
		public string Id_user {
            get { return getAttrVal<string>("Id_user",null); }
            set { setAttrVal<string>("Id_user", value); }
        }
		public bool? Personal {
            get { return getAttrVal<FBoolean>("Personal",null); }
            set { setAttrVal<FBoolean>("Personal", value); }
        }
		public bool? Administrative {
            get { return getAttrVal<FBoolean>("Administrative",null); }
            set { setAttrVal<FBoolean>("Administrative", value); }
        }
		public int? Attribute {
            get { return getAttrVal<int?>("Attribute",null); }
            set { setAttrVal<int?>("Attribute", value); }
        }
		public string Des {
            get { return getAttrVal<string>("Des",null); }
            set { setAttrVal<string>("Des", value); }
        }
		public string Createdby {
            get { return getAttrVal<string>("Createdby",null); }
            set { setAttrVal<string>("Createdby", value); }
        }
		public DateTime? Createdtime {
            get { return getAttrVal<FDateTime>("Createdtime",null); }
            set { setAttrVal<FDateTime>("Createdtime", value); }
        }
		public string Modifiedby {
            get { return getAttrVal<string>("Modifiedby",null); }
            set { setAttrVal<string>("Modifiedby", value); }
        }
		public DateTime? Modifiedtime {
            get { return getAttrVal<FDateTime>("Modifiedtime",null); }
            set { setAttrVal<FDateTime>("Modifiedtime", value); }
        }
		public string Org_code {
            get { return getAttrVal<string>("Org_code",null); }
            set { setAttrVal<string>("Org_code", value); }
        }
		public string Org_name {
            get { return getAttrVal<string>("Org_name",null); }
            set { setAttrVal<string>("Org_name", value); }
        }
		public string Dept_code {
            get { return getAttrVal<string>("Dept_code",null); }
            set { setAttrVal<string>("Dept_code", value); }
        }
		public string Dept_name {
            get { return getAttrVal<string>("Dept_name",null); }
            set { setAttrVal<string>("Dept_name", value); }
        }
		public string User_name {
            get { return getAttrVal<string>("User_name",null); }
            set { setAttrVal<string>("User_name", value); }
        }
		public string User_code {
            get { return getAttrVal<string>("User_code",null); }
            set { setAttrVal<string>("User_code", value); }
        }
        public int Ds {
            get { return getAttrVal<int>("Ds",0);}
            set { setAttrVal<int>("Ds", value); }
        }

        public DateTime? Sv        {
            get { return getAttrVal<FDateTime>("Sv",null); }
            set { setAttrVal<FDateTime>("Sv", value); }
        }
        
        /// <summary>
        /// 返回表名
        /// </summary>
        /// <returns></returns>
        public override string getTableName()
        {
            return "ci_mr_knowledge_type";
        }
        
        /// <summary>
        /// 返回表别名
        /// </summary>
        /// <returns></returns>
        public override string getAliasTableName()
        {
            return "a0";
        }
        
        /// <summary>
        /// 返回主键字段名
        /// </summary>
        /// <returns></returns>
        public override string getPKFieldName()
        {
            return "Id_knowledge_type";
        }

        /// <summary>
        /// 返回全路径DO类名
        /// </summary>
        /// <returns></returns>
        public override string getFullClassName()
        {
            return "iih.ci.mr.knowledgetype.d.KnowledgeTypeDO";
        }

        /// <summary>
        /// 返回属性列表
        /// </summary>
        /// <returns></returns>
        public override string[] getAttrNames()
        {
            return new string[]{"Ds","Sv",  "Id_knowledge_type", "Id_grp", "Id_org", "Sortno", "Code", "Name", "Pycode", "Wbcode", "Mnecode", "Id_dept", "Id_user", "Personal", "Administrative", "Attribute", "Des", "Createdby", "Createdtime", "Modifiedby", "Modifiedtime", "Org_code", "Org_name", "Dept_code", "Dept_name", "User_name", "User_code"};
        }
        
		/// <summary>
        /// 设置IBDataInfo接口映射数据
        /// </summary>
        /// <returns></returns>
		protected override void setBdDataInfoNameMap() {
			if (base.name_path_map == null)
            {
                base.name_path_map = new Dictionary<string, string>();
				base.name_path_map.Add("id","Id_knowledge_type");
				base.name_path_map.Add("id_org","Id_org");
				base.name_path_map.Add("code","Code");
				base.name_path_map.Add("name","Name");
				base.name_path_map.Add("id_group","Id_grp");
				base.name_path_map.Add("pycode","Pycode");
				base.name_path_map.Add("wbcode","Wbcode");
				base.name_path_map.Add("mnecode","Mnecode");
            }
		}
    }
}
